tests/tests_insights_ansible_host.yaml

```diff
@@ -0,0 +1,116 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Insights ansible host test
+  hosts: all
+  gather_facts: false
+  become: true
+
+  tasks:
+    - name: Setup Insights
+      import_tasks: tasks/setup_insights.yml
+
+    - name: Test ansible_host
+      block:
+        - name: Add ansible_host and register insights
+          include_role:
+            name: linux-system-roles.rhc
+            public: true
+          vars:
+            rhc_auth:
+              login:
+                username: "{{ lsr_rhc_test_data.reg_username }}"
+                password: "{{ lsr_rhc_test_data.reg_password }}"
+            rhc_insights:
+              ansible_host: host
+              remediation: absent
+              state: present
+            rhc_organization: "{{ lsr_rhc_test_data.reg_organization }}"
+            rhc_server:
+              hostname: "{{ lsr_rhc_test_data.candlepin_host }}"
+              port: "{{ lsr_rhc_test_data.candlepin_port }}"
+              prefix: "{{ lsr_rhc_test_data.candlepin_prefix }}"
+              insecure: "{{ lsr_rhc_test_data.candlepin_insecure }}"
+            rhc_baseurl: "{{ lsr_rhc_test_data.baseurl | d(omit) }}"
+
+        - name: Check ansible_host is set to 'host' in config file
+          command:
+            grep -ixq "^ansible_host=host" {{ __rhc_insights_conf }}
```

Comment:
> Can’t lineinfile be used here, rather than a grep command?

Yes, if there is a substantial benefit in terms of code readability and long term maintainability.  Since this is test code (as opposed to code in the `tasks/` runtime directory), I'm willing to allow a bit more leniency towards the preference of the original developer.

Reply:
> I’d rather consider single quoted string. Although there is no interpolation or control characters, it would convey a message of an exact string.

The general Ansible style is to only quote things that require quoting, so I would prefer no quotes at all.

Reply:
> Although the Insights Client config path is constant and doesn’t contain any special characters or spaces, paths can contain those. Paths may contain single quotes too, which can’t be properly escaped at all. It may make sense to rather store it in an enviornment variable.

Not sure what you mean by environment variable here - where would the value come from?  How would it be set for the task?  That seems unnecessarily complicated.  Note that Ansible has a built-in filter for quoting variables expanded in a shell command - `quote` - so this should correctly be `{{ __rhc_insights_conf | quote }}` - maybe with the environment variable comment you were getting at the fact that using any sort of single or double quoting is not guaranteed to work e.g. `'{{ __rhc_insights_conf }}'` will fail if `__rhc_insights_conf` (perversely) contains single quote characters.  See https://docs.ansible.com/ansible/2.9/user_guide/playbooks_filters.html#id8

Reply:
If there is `| quote`, which I didn’t know about, there is no need for something as complicated as an environment variable. 👍🏻  Cool!